sympy/tensor/tests/test_indexed.py

```diff
@@ -229,7 +229,21 @@ def test_IndexedBase_assumptions_inheritance():
 
     assert I_inherit.is_integer
     assert I_explicit.is_integer
-    assert I_inherit == I_explicit
+    assert I_inherit.label.is_integer
+    assert I_explicit.label.is_integer
+    assert I_explicit == I_inherit
+
+
+def test_IndexedBase_no_upcasting_of_label():
```

Comment:
A better name can be `def test_issue_17652`.

Reply:
:+1: